sdk/src/cose_sign.rs

```diff
@@ -15,35 +15,18 @@ use crate::time_stamp::{cose_timestamp_countersign, make_cose_timestamp};
 use crate::{Error, Result, Signer}; // enable when TimeStamp Authority is ready
 
 use ciborium::value::Value;
-use coset::{iana, CoseSign1, CoseSign1Builder, HeaderBuilder, Label, TaggedCborSerializable};
+use coset::{
+    iana, CoseSign1, CoseSign1Builder, Header, HeaderBuilder, Label, TaggedCborSerializable,
+};
 
-/// Returns signed Cose_Sign1 bytes for "data".  The Cose_Sign1 will be signed with the algorithm from `Signer`.
-pub fn cose_sign(signer: &dyn Signer, data: &[u8], box_size: usize) -> Result<Vec<u8>> {
-    // 13.2.1. X.509 Certificates
-    //
-    // X.509 Certificates are stored in a header named x5chain draft-ietf-cose-x509.
-    // The value is a CBOR array of byte strings, each of which contains the certificate
-    // encoded as ASN.1 distinguished encoding rules (DER). This array must contain at
-    // least one element. The first element of the array must be the certificate of
-    // the signer, and the subjectPublicKeyInfo element of the certificate will be the
-    // public key used to validate the signature. The Validity member of the TBSCertificate
-    // sequence provides the time validity period of the certificate.
-
-    /*
-       This header parameter allows for a single X.509 certificate or a
-       chain of X.509 certificates to be carried in the message.
-
-       *  If a single certificate is conveyed, it is placed in a CBOR
-           byte string.
-
-       *  If multiple certificates are conveyed, a CBOR array of byte
-           strings is used, with each certificate being in its own byte
-           string.
-    */
-
-    let alg = signer.alg().ok_or(Error::UnsupportedType)?;
-
-    let alg_id = match alg.as_ref() {
+fn build_unprotected_header(
+    data: &[u8],
+    alg: &str,
+    certs: Vec<Vec<u8>>,
+    ta_url: Option<String>,
```

Comment:
Granted, this is an internal API, but is there any reason for this to consume the `String`? Could this be `Option<&str>` instead?

Reply:
This was marked resolved, but I don't see any comment or code change that would resolve it.